### PR TITLE
Fix unhandled exception on initialization

### PR DESCRIPTION
**Pull request by emersonsiega**
Refers to #49 

Changed `initialise` method to simply rethrow exceptions that may occur during video initialization.

This allows users to know it there's an error during the initialization method (listener is just added if controller is correctly initialized).

To reproduce, just load a network/vimeo/youtube/asset that doesn't exists.

### CODE REVIEW

**Comment by newtaDev** [comment]
Thanks for the PR ❤️ I really appreciate your effort 🤗
I will go through this and will merge ASAP

File: lib/src/controllers/pod_player_controller.dart [hunk 1]
```diff
@@ -35,17 +39,24 @@ class PodPlayerController {
...
+      try {
+        if (!_isCtrInitialised) {
+          await _ctr.videoInit();
+          podLog('$getTag Pod player Initialized');
+        } else {
+          podLog('$getTag Pod Player Controller Already Initialized');
+        }
+      } catch (error) {
+        podLog('$getTag Pod Player Controller failed to initialize');
+        _initializationError = error;
```
Comment (newtaDev) [inline lib/src/controllers/pod_player_controller.dart:59]: Is it necessary to have `_initializationError`  variable....?
instead, we can rethrow the error directly right?
```dart
  } catch (error) {
        podLog('$getTag Pod Player Controller failed to initialize');
        rethrow;
      }
```

**Comment by newtaDev** [comment]
> @newtaDev and there's more coming (probably today)! hahaha
> 
> thanks!

@emersonsiega 
wow 🤩 
Very Happy to have a contributor like you, Tanks a lot...🤗

File: lib/src/controllers/pod_player_controller.dart (hunk 1 above)
Comment (newtaDev) [reply lib/src/controllers/pod_player_controller.dart:59]: Please verify this so I can merge changes

Comment (emersonsiega) [reply lib/src/controllers/pod_player_controller.dart:59]: Actually, this is the main problem... The rethrow doesn't throws the error to the outside caller, cause it's inside a addPostFrameCallback... 
Moving the responsibility of throwing the error to the `_checkAndWaitTillInitialized` solves this problem

Comment (newtaDev) [reply lib/src/controllers/pod_player_controller.dart:59]: Hoo, I was not aware of that Thank you ❤️

**Comment by newtaDev** [comment]
Already merged ✌️
Thank you for contributing ❤️

**Pull request closed without merging**